src/compiler/binder.ts

```diff
@@ -173,6 +173,14 @@ namespace ts {
             return node.name ? declarationNameToString(node.name) : getDeclarationName(node);
         }
 
+        /**
+         * Declares a Symbol for the node and adds it to symbols. Reports errors for conflicting identifier names.
+         * @param symbolTable - The symbol table which node will be added to.
+         * @param parent - If node is in a class, parent denotes the parent declaration.

```

Comment:
No, it's for any member relationship. So it can be in a class, interface, object type literal, enum or module (if it's a module the member has to be exported to have a parent).